fl4health/servers/nnunet_server.py

```diff
@@ -138,43 +153,27 @@ def initialize_server_model(self) -> None:
 
         self.server_model = model
 
-    def fit(self, num_rounds: int, timeout: Optional[float]) -> Tuple[History, float]:
-        """
-        Same as parent method except initialize hook method is called first
-        """
-        # Initialize the server
-        if not self.initialized:
-            self.initialize(server_round=0, timeout=timeout)
-
-        return FlServerWithCheckpointing.fit(self, num_rounds, timeout)
-
-    def initialize(self, server_round: int, timeout: Optional[float] = None) -> None:
+    def update_before_fit_round(self, server_round: int, timeout: Optional[float] = None) -> None:
```

Comment:
Rather than overridding an initialize function, we use a more general hook that is called every time the server calls `fit_round`. This function only does anything on the very first server round.